File: practice-algos/sock_merchant/gpt_sol.py
def sockMerchant(n, ar):
    color_counts = {}
    for color in ar:
        if color in color_counts:
            color_counts[color] += 1
        else:
            color_counts[color] = 1

    return sum(count // 2 for count in color_counts.values())

# Example usage
n = 7
ar = [1, 2, 1, 2, 3, 1, 2]
print(sockMerchant(n, ar))  # Output: 2

File: practice-algos/sock_merchant/gpt_sol.js
function sockMerchant(n, ar) {
    const colorCounts = {};
    ar.forEach(color => {
        colorCounts[color] = (colorCounts[color] || 0) + 1;
    });

    let pairs = 0;
    Object.values(colorCounts).forEach(count => {
        pairs += Math.floor(count / 2);
    });

    return pairs;
}

// Example usage
const n = 7;
const ar = [1, 2, 1, 2, 3, 1, 2];
console.log(sockMerchant(n, ar));  // Output: 2
